contrib/windows/GLPI-AgentMonitor/GLPI-AgentMonitor.cpp

```diff
@@ -0,0 +1,688 @@
+    }
+    else
+        LoadString(hInst, IDS_ERR_NOTRUNNING, szAgStatus, dwAgStatusLen);
+}
+
+// Requests an inventory via HTTP
+VOID ForceInventory(HWND hWnd)
+{
+    if (running)
+    {
+        if (glpiAgentOk)
+        {
+            HINTERNET hReq = WinHttpOpenRequest(hConn, L"GET", L"/now", NULL, WINHTTP_NO_REFERER, WINHTTP_DEFAULT_ACCEPT_TYPES,
+                WINHTTP_FLAG_BYPASS_PROXY_CACHE);
+
+            if (WinHttpSendRequest(hReq, WINHTTP_NO_ADDITIONAL_HEADERS, NULL, WINHTTP_NO_REQUEST_DATA, NULL, NULL, NULL)) 
+            {
+                if (WinHttpReceiveResponse(hReq, NULL)) {
+                    DWORD dwStatusCode = 0;
+                    DWORD dwSize = sizeof(dwStatusCode);
+
+                    if (!WinHttpQueryHeaders(hReq, WINHTTP_QUERY_STATUS_CODE | WINHTTP_QUERY_FLAG_NUMBER,
+                        WINHTTP_HEADER_NAME_BY_INDEX, &dwStatusCode, &dwSize, WINHTTP_NO_HEADER_INDEX))
+                    {
+                        LoadString(hInst, IDS_ERR_FORCEINV_NORESPONSE, szBuffer, dwBufferLen);
+                        LoadString(hInst, IDS_ERROR, szTitleBuf, dwTitleBufLen);
+                        MessageBox(hWnd, szBuffer, szTitleBuf, MB_OK | MB_ICONERROR);
+                    }
+                    else {
+                        if (dwStatusCode != 200)
+                        {
+                            LoadString(hInst, IDS_ERR_FORCEINV_NOTALLOWED, szBuffer, dwBufferLen);
+                            LoadString(hInst, IDS_ERROR, szTitleBuf, dwTitleBufLen);
+                            MessageBox(hWnd, szBuffer, szTitleBuf, MB_OK | MB_ICONERROR);
+                        }
+                        else
+                        {
+                            LoadString(hInst, IDS_MSG_FORCEINV_OK, szBuffer, dwBufferLen);
+                            LoadString(hInst, IDS_APP_TITLE, szTitleBuf, dwTitleBufLen);
+                            MessageBox(hWnd, szBuffer, szTitleBuf, MB_OK | MB_ICONINFORMATION);
+                        }
+                    }
+                }
+            }
+            WinHttpCloseHandle(hReq);
+        }
+        else
+        {
+            LoadString(hInst, IDS_ERR_AGENTERR, szBuffer, dwBufferLen);
+            LoadString(hInst, IDS_ERROR, szTitleBuf, dwTitleBufLen);
+            MessageBox(hWnd, szBuffer, szTitleBuf, MB_OK | MB_ICONERROR);
+        }
+    }
+    else 
+    {
+        LoadString(hInst, IDS_ERR_NOTRUNNING, szBuffer, dwBufferLen);
+        LoadString(hInst, IDS_ERROR, szTitleBuf, dwTitleBufLen);
+        MessageBox(hWnd, szBuffer, szTitleBuf, MB_OK | MB_ICONERROR);
+    }
+}
+
+// Updates the main window statuses
+VOID CALLBACK UpdateStatus(HWND hWnd, UINT message, UINT idTimer, DWORD dwTime)
+{
+    SC_HANDLE hSc = OpenSCManager(NULL, SERVICES_ACTIVE_DATABASE, SERVICE_QUERY_STATUS);
+    SC_HANDLE hSvc = OpenService(hSc, SERVICE_NAME, SERVICE_QUERY_STATUS);
+    SERVICE_STATUS lpsvcStatus;
+    if (!QueryServiceStatus(hSvc, &lpsvcStatus)) { }
```

Comment:
I don't see why a test to run an empty bloc would be required here. Did you construct it to eventually report a problem or just to abort in the case we don't get the service ?
```suggestion
    QueryServiceStatus(hSvc, &lpsvcStatus);
```